faker/providers/file/__init__.py

```diff
@@ -269,12 +270,28 @@ def file_path(
         If ``absolute`` is ``True`` (default), the generated path starts with
         ``/`` and is absolute. Otherwise, the generated path is relative.
 
+        If used, ``extension`` can be either a string, forcing that extension, a
+        list of strings (one will be picked at random), or an empty list (the
+        path will have no extension). Default behaviour is the same as |file_name|
+
         :sample: size=10
         :sample: depth=3
         :sample: depth=5, category='video'
         :sample: depth=5, category='video', extension='abcdef'
+        :sample: extension=[]
+        :sample: extension=["a", "bc", "def"]
         """
+        no_extension = False
+        if isinstance(extension, list):
+            if extension:
+                extension = self.random_element(extension)
+            else:
+                extension = None
+                no_extension = True
+
         file: str = self.file_name(category, extension)
+        if no_extension:
+            file = re.sub(r"\.[^/]+$", "", file)
```

Comment:
If we have to strip, I'd rather use `rsplit()`:

```suggestion
            file = file.rsplit(".", 1)[0]
```